File: src/WebUI/src/models/game-server-stats.ts
import type { GameMode } from './game-mode'
import type { Region } from './region'

interface GameStats {
  playingCount: number
}

export type GameServerModeStats = Partial<Record<GameMode, GameStats>>

export type GameServerRegionStats = Partial<Record<Region, GameServerModeStats>>

export interface GameServerStats {
  total: GameStats
  regions: GameServerRegionStats
}

File: src/Application/Games/Models/GameServerStats.cs
using Crpg.Domain.Entities;
using Crpg.Domain.Entities.Servers;

namespace Crpg.Application.Games.Models;

public record GameServerStats
{
    public GameStats Total { get; set; } = default!;
    public Dictionary<Region, Dictionary<GameMode, GameStats>> Regions { get; set; } = default!;
}

public record GameStats
{
    /// <summary>Number of users currently playing on servers.</summary>
    public int PlayingCount { get; set; }
}
